### PR TITLE
Create macOS .pkg pipeline

### PR DESCRIPTION
**Pull request by cpyarger**
### Description
This PR adds to the CI pipeline unsigned macOS packaging

#### Old Behavior

no .pkg file 
#### New Behavior

oooh shiny installer

### CODE REVIEW

**Comment by cpyarger** [comment]
This also fixes an issue with mac builds, Brew has updated to QT6 and as a result qt is no longer installed in the normal directory,

**Comment by Xaymar** [comment]
The pkgproj file should be a templated file, instead of always pointing at a fixed path.

**Comment by cpyarger** [comment]
> The pkgproj file should be a templated file, instead of always pointing at a fixed path.

no clue how to do that, this is how I learned to do it from obs-websocket

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -195,8 +196,10 @@ jobs:
     - name: "Prerequisties: Qt"
       shell: bash
       run: |
-        brew install qt5
-        ls -lha /usr/local/opt/qt/lib/cmake/Qt5
+        brew install qt@5
+        export LDFLAGS="-L/usr/local/Cellar/qt@5/5.15.2/lib"
+        export CPPFLAGS="-I/usr/local/Cellar/qt@5/5.15.2/include"
+        ls -lha /usr/local/Cellar/qt@5/5.15.2/lib/cmake/Qt5
```
Comment (Xaymar) [inline .github/workflows/main.yml:202]: I'm unsure what this change does, qt5 should point at Qt5, not Qt6. This is even listed on the homebrew site itself as an alias to qt@5: https://formulae.brew.sh/formula/qt@5

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -205,17 +208,34 @@ jobs:
...
+          -DQt5_DIR=/usr/local/Cellar/qt@5/5.15.2/lib/cmake/Qt5
     - name: "Build Project"
       shell: bash
       run: |
         cmake --build "build/temp" --config RelWithDebInfo --target install
+    - name: 'Install prerequisite: Packages app'
+      if: success()
+      shell: bash
+      run: |
+        curl -L -O https://s3-us-west-2.amazonaws.com/obs-nightly/Packages.pkg
```
Comment (Xaymar) [inline .github/workflows/main.yml:220]: Is it possible to do this without third-party dependencies? _Especially_ third party references to binaries uploaded by OBS Project?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 name: Build
 
-on:
```
Comment (Xaymar) [inline .github/workflows/main.yml:3]: Formatting fixes should be a separate commit, this touches way too much for a single commit.

File: .github/workflows/main.yml (hunk 1 above)
Comment (cpyarger) [reply .github/workflows/main.yml:202]: check the paths, qt5 breaks, qt@5 doesn't

File: .github/workflows/main.yml (hunk 2 above)
Comment (cpyarger) [reply .github/workflows/main.yml]: I am sure that there is, but I do not know it.

**Comment by Xaymar** [comment]
I've split this PR into two parts. This one will stay for the .pkg step, which I'll have to adjust somewhat, and another which will fix CI.

**Comment by cpyarger** [comment]
Sounds Good,

**Comment by Xaymar** [comment]
Trying to figure out how to affect the name of the .pkg file without affecting the name of the actual installation, sadly the .pkgproj file format is undocumented. I did find documentation about the XML format for the .pkg information file, but it's not exactly applicable for this.

**Review by Xaymar — APPROVED** [review]

**Comment by Xaymar** [comment]
After a bit of testing and writing my own tool to create .pkgproj files, I now have a working structure. Waiting for CI to give the green light, and then merging if everything is working as expected.

**Pull request merged**